### PR TITLE
Add API to query for responses

### PR DESCRIPTION
**Pull request by brettwold**
Allows responses to be queried for all domains

### CODE REVIEW

File: flow-base-api/src/main/java/com/aevi/sdk/flow/model/ResponseQuery.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    /**
+     * An internal method used by this API to set the response type required for this query.
+     *
+     * FOR INTERNAL USE ONLY: This method should not be used by external classes
+     *
+     * @param responseType The class name of the response type to expect for this query
+     */
+    public void setResponseType(String responseType) {
```
Comment (andreas-borglin) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/model/ResponseQuery.java]: minor, but maybe put at bottom of class to make it less visible

File: flow-base-api/src/main/java/com/aevi/sdk/flow/model/ResponseQuery.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        this.responseType = responseType;
+    }
+
+    public String getResponseType() {
```
[the remark this answers is not in the document]
Comment (brettwold) [reply flow-base-api/src/main/java/com/aevi/sdk/flow/model/ResponseQuery.java]: No they don't only if we say they do...And if they do then there's plenty of other public methods in this API that do not have javadocs (including this package even).

File: PaymentInitiationSample/src/main/java/com/aevi/sdk/pos/flow/paymentinitiationsample/ui/adapter/PaymentResponsesAdapter.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    protected void onBindViewHolderToObject(BaseTwoLineAdapter.ViewHolder holder, PaymentResponse paymentResponse, int i) {
+        Context context = holder.title.getContext();
+        Amounts amounts = paymentResponse.getTotalAmountsProcessed();
+        holder.title.setText(context.getString(R.string.payment_response_title,
+                                               paymentResponse.getOriginatingPayment().getFlowType().toUpperCase(),
+                                               paymentResponse.getOutcome().name()));
+        holder.subtitle.setText(context.getString(R.string.payment_response_totals,
+                                                  AmountFormatter.formatAmount(amounts.getCurrency(), amounts.getBaseAmountValue()),
+                                                  AmountFormatter.formatAmount(amounts.getCurrency(),
+                                                                               amounts.getAdditionalAmountValue(AmountIdentifiers.AMOUNT_TIP)),
```
[the remark this answers is not in the document]
Comment (brettwold) [reply PaymentInitiationSample/src/main/java/com/aevi/sdk/pos/flow/paymentinitiationsample/ui/adapter/PaymentResponsesAdapter.java]: To do anything else in this sample would require a better UI. Its only a sample and the payment initiator doesn't allow other additional amounts anyway

[the remark this answers is not in the document]
Comment (andreas-borglin) [reply PaymentInitiationSample/src/main/java/com/aevi/sdk/pos/flow/paymentinitiationsample/ui/adapter/PaymentResponsesAdapter.java]: Sure, the initiatior doesn't, but the Flow Service Sample does. If we ain't gonna show them "correctly", might as well just not show additionals and just show total and keep it simpler IMO?

**Pull request merged**